src/main/java/net/dv8tion/jda/internal/handle/InviteCreateHandler.java

```diff
@@ -74,7 +74,31 @@ protected Long handleInternally(DataObject content)
         InviteImpl.ChannelImpl channel = new InviteImpl.ChannelImpl(realChannel);
         InviteImpl.GuildImpl guild = new InviteImpl.GuildImpl(realGuild);
 
-        Invite invite = new InviteImpl(getJDA(), code, expanded, inviter, maxAge, maxUses, temporary, creationTime, 0, channel, guild, null, Invite.InviteType.GUILD);
+        final Invite.TargetType targetType = Invite.TargetType.fromId(content.getInt("target_type", -1));
+        final User targetUser;
+        final InviteImpl.EmbeddedApplicationImpl application;
+
+        switch (targetType)
+        {
+        case STREAM:
+            DataObject targetUserObject = content.getObject("target_user");
+            targetUser = getJDA().getEntityBuilder().createUser(targetUserObject);
+            application = null;
+            break;
+        case EMBEDDED_APPLICATION:
+            DataObject applicationObject = content.getObject("target_application");
+            application = new InviteImpl.EmbeddedApplicationImpl(
+                    applicationObject.getString("icon", null), applicationObject.getString("name"), applicationObject.getString("description"),
+                    applicationObject.getString("summary"), applicationObject.getLong("id"), applicationObject.getInt("max_participants", -1)
+            );
+            targetUser = null;
+            break;
+        default:
+            application = null;
+            targetUser = null;
```

Comment:
```suggestion
            case STREAM:
                DataObject targetUserObject = content.getObject("target_user");
                targetUser = getJDA().getEntityBuilder().createUser(targetUserObject);
                application = null;
                break;
            case EMBEDDED_APPLICATION:
                DataObject applicationObject = content.getObject("target_application");
                application = new InviteImpl.EmbeddedApplicationImpl(
                        applicationObject.getString("icon", null), applicationObject.getString("name"), applicationObject.getString("description"),
                        applicationObject.getString("summary"), applicationObject.getLong("id"), applicationObject.getInt("max_participants", -1)
                );
                targetUser = null;
                break;
            default:
                application = null;
                targetUser = null;
```

Reply:
I didn't indent these cases as they weren't indented in other switches I saw (eg in the EntityBuilder class) and the indention for them are set to false in the editorconfig. Does it make sense to indent them?